### PR TITLE
309 fix local chain test timeout issues

### PR DESCRIPTION
**Pull request by gabrielfior**

### CODE REVIEW

File: tests_integration/README.md [hunk 1]
```diff
@@ -1,6 +1,10 @@
 # Integration tests
 
-These tests use a local chain to enable the accurate testing of functions that execute transactions on the blockchain.
+These tests use a local chain (powered by the ![ape-test](https://docs.apeworx.io/ape/stable/userguides/testing.html) pytest plugin) to enable the accurate testing of functions that execute transactions on the blockchain.
+
+Note that the plugin, regardless of test logic, connects to the RPC of the default ecosystem provider. Hence, tests which do not require a local chain should not be pasted in this folder.
+
+Also note that it's mandatory to use the fixture `local_web3` inside every test inside this folder, otherwise the test fails due to an issue with ape (see comment inside fixture).
```
Comment (evangriffiths) [inline tests_integration/README.md:7]: This bit isn't clear to me. Can you explain?

Could we move all the tests that require it to a subdirectory so we don't have to do this? e.g.
```
% tree
.
└── tests_integration
    ├── test_local_web3
    │   └── conftest.py
    └── test_something_that_doesnt_need_local_web3.py
```

Comment (gabrielfior) [reply tests_integration/README.md]: I created an additional directory (`tests_integration_with_local_chain`) - I hope that fixes this.

File: pyproject.toml [hunk 2]
```diff
@@ -37,12 +37,15 @@ google-api-python-client = { version = "2.95.0", optional = true}
...
 safe-eth-py = "^6.0.0b14"
-eth-account = "^0.8.0"
+eth-account = "^0.11.2"
 prompt-toolkit = "^3.0.43"
 safe-cli = "^1.0.0"
 langfuse = "^2.27.1"
 openai = { version = "^1.0.0", optional = true}
-base58 = "^2.1.1"
+base58 = ">=1.0.2,<2.0"
+eth-ape = "^0.8.10"
```
[the remark this answers is not in the document]
Comment (gabrielfior) [reply pyproject.toml:46]: moved

**Comment by gabrielfior** [comment]
@evangriffiths note that it wasn't possible to move `ape-config.yaml` to the tests folder because one cannot pass the config file as parameter (see [docs](https://docs.apeworx.io/ape/stable/userguides/config.html#configure-ape)) :(

File: pyproject.toml [hunk 3]
```diff
@@ -37,12 +37,15 @@ google-api-python-client = { version = "2.95.0", optional = true}
...
+cowswap-python-client ="^0.1.1"
```
[the remark this answers is not in the document]
Comment (evangriffiths) [reply pyproject.toml:46]: How come you don't want to remove?

File: tests_integration_with_local_chain/markets/omen/test_local_chain.py [hunk 4]
```diff
@@ -90,7 +89,9 @@ def test_send_xdai_from_locked_account(
     "address, expected",
     [
         (
-            Web3.to_checksum_address(get_anvil_test_accounts()[0].address),
+            Web3.to_checksum_address(
+                "0xf39Fd6e51aad88F6F4ce6aB8827279cffFb92266"
```
Comment (evangriffiths) [inline tests_integration_with_local_chain/markets/omen/test_local_chain.py:93]: Should this be `accounts[0]`?

File: tests_integration/local_chain_utils.py [hunk 5]
```diff
@@ -1,154 +0,0 @@
-import atexit
```
Comment (evangriffiths) [inline tests_integration/local_chain_utils.py:1]: has this file not been `git rm`d?

File: prediction_market_agent_tooling/monitor/monitor.py [hunk 6]
```diff
@@ -229,7 +229,8 @@ def monitor_agent(agent: DeployedAgent) -> None:
         )
         .interactive()
     )
-    st.altair_chart(
+
+    st.altair_chart(  # type: ignore
```
Comment (evangriffiths) [inline prediction_market_agent_tooling/monitor/monitor.py:233]: how come these unrelated files need typing changes?

Comment (gabrielfior) [reply prediction_market_agent_tooling/monitor/monitor.py:233]: Slack discussion - mypy version changes account for this

File: pyproject.toml (hunk 3 above)
[the remark this answers is not in the document]
Comment (gabrielfior) [reply pyproject.toml:46]: Ops, might have messed up the merge - removed now!

File: tests_integration_with_local_chain/markets/omen/test_local_chain.py (hunk 4 above)
Comment (gabrielfior) [reply tests_integration_with_local_chain/markets/omen/test_local_chain.py:93]: In the latest version, I added `0xf39Fd6e51aad88F6F4ce6aB8827279cffFb92266`, which is Anvil's hardcoded 1st account.
This is not optimal, but I don't see a great other option - do you?
For context, using `accounts` would be nice, but using `pytest.parametrize` with fixtures has some challenges and generally involve using `pytest.request.fixturevalue("my-fixture")`, so also not ideal.

File: tests_integration/local_chain_utils.py (hunk 5 above)
Comment (gabrielfior) [reply tests_integration/local_chain_utils.py:1]: This has been deleted.

File: tests_integration_with_local_chain/markets/omen/test_local_chain.py (hunk 4 above)
Comment (evangriffiths) [reply tests_integration_with_local_chain/markets/omen/test_local_chain.py:93]: Is `0x184ca44A6c3cfc05bCF7246ac14101Ddb9423eAa` also an anvil wallet address, or just some random address?

Are we testing here that our anvil address has some xdai, but this random address `0x184c...` doesn't? Sorry, it's not clear to me. If this is the case, what about

```
def test_is_minimum_required_balance(local_web3: Web3, accounts: list[TestAccount]) -> None:
    min_balance = xdai_type(0.5)

    # Our anvil accounts are funded
    for account in accounts:
        assert is_minimum_required_balance(account.address, min_balance, local_web3)

    # A new account should not have the minimum balance
    a_new_address = local_web3.eth.account.create().address
    assert not is_minimum_required_balance(a_new_address, min_balance, local_web3)
```

I don't really mind, as long as it's clear what these magic addresses are!

Comment (gabrielfior) [reply tests_integration_with_local_chain/markets/omen/test_local_chain.py]: Refactored tests

**Review by evangriffiths — APPROVED** [review]
